src/precice/impl/Participant.hpp

```diff
@@ -130,22 +132,22 @@ class Participant {
   /** Provides access to write \ref DataContext
    * @remarks does not contain nullptr.
    */
-  const utils::ptr_vector<DataContext> &writeDataContexts() const;
+  const std::vector<WriteDataContext *> &writeDataContexts() const;
```

Comment:
I am not sure about this change. I agree the old one is ugly, but why not a vector of references?
@fsimonis typically has a good opinion on such things.

More similar occasions below.

Reply:
IIRC the main reasons for moving from `utils::ptr_vector<T>` to `std::vector<T>` were:

1. some standard functionality that I needed for the implementation was missing and I did not want to implement it in `utils::ptr_vector`, if `std::vector` can also do the job.
2. I do not understand why `utils::ptr_vector<T>` actually exists. It looks to me just like `std::vector` without a real benefit.

I can also try to get a vector of references working :+1:

Reply:
Also done in ce0eb84. I will keep this one open until @fsimonis has reviewed.

Reply:
> I am not sure about this change. I agree the old one is ugly, but why not a vector of references?

References are immutable. Imagine what `push_back` means if the new element cannot be assigned. So, this doesn't work.
Raw pointers are not a problem as long as they are just used to access the pointees. Raw owning pointers are the real problem.

> I do not understand why utils::ptr_vector<T> actually exists.

The `utils::ptr_vector<T>` is a _possibly owning_ vector of pointers to objects.
I already replaced quite some occurrences with normal vectors.

> It looks to me just like std::vector without a real benefit.

It allows to delete all contained elements. That's about it.

@BenjaminRodenberg Maybe we can implement this as `std::vector<WriteDataContext>` instead? Who owns these Contexts?